src/app/tasks/tasks-list/task-item/directives/drag.directive.ts

```diff
@@ -0,0 +1,90 @@
+import { Directive, ElementRef, inject, Renderer2, Input, OnInit, HostListener } from '@angular/core';
+
+export interface ElemActions {
+  onDrag?: () => void;
+  onClick?: () => void;
+}
+
+@Directive({
+  selector: '[appDrag]',
+  standalone: true
+})
+export class DragDirective implements OnInit {
+  private el: ElementRef = inject(ElementRef);
+  private renderer: Renderer2 = inject(Renderer2);
+
+  private element!: any;
+  private startX!: number;
+  private isDragged!: boolean;
+  private holdDuration!: number;    
+  private timeout!: NodeJS.Timeout;
+
+  @Input() appDrag?: ElemActions;
+
+  public ngOnInit(): void {
+    this.initFields();
+
+    this.renderer.setStyle(this.element,'touch-action','pan-y');
+  }
+
+  @HostListener('pointerdown', ['$event']) onPointerDown(event: PointerEvent) {
+    const clientX = event.clientX;
+
+    this.startX = clientX;
+    this.timeout = setTimeout(() => {
+      this.isDragged = true;
+    }, this.holdDuration);
+  }
+  
+  @HostListener('pointermove', ['$event']) onPointerMove(event: PointerEvent) {
+    if (this.isDragged) {
+      const clientX = event.clientX;
+      const delta = clientX - this.startX;
+
+      this.renderer.removeStyle(this.element,'transition');
+
+      this.transform(`${delta}px`);
+    }
+  }
+
+  @HostListener('pointerup', ['$event']) onPointerUp(event: PointerEvent) {
```

Comment:
Let's improve readability and move all the relevant code to `handleDrag`.

```
  @HostListener('pointerup', ['$event']) onPointerUp(event: PointerEvent) {
    clearTimeout(this.timeout);

    this.isDragged ? this.handleDrag(event) : this.handleClick(); 

  }
```

Reply:
got it.

i think i can even pass only event.clientX to handleDrag since i dont really need the whole event object in this function. what do you say?